web/lib/components/flashcards/PageAllSets.jsx

```diff
@@ -1,32 +1,63 @@
-import { useState } from 'react'
+import { useState, useContext } from 'react'
 import { useRouter } from 'next/router'
 
-import { useUser } from '@/lib/auth'
-import { useTranslation } from '@/lib/i18n'
-
+import { Input } from '../common/Input'
+import { Button } from '../common/Button'
+import { Dialog } from '../common/Dialog'
 import { Layout } from '../common/Layout'
 import { Breadcrumb } from '../common/Breadcrumb'
-import { Button } from '../common/Button'
-import { ArrowLeftIcon, ArrowRightIcon } from '../common/Icons'
+import { ArrowLeftIcon, ArrowRightIcon, PlusIcon } from '../common/Icons'
+
+import { useUser } from '../../auth'
+import { useTranslation } from '../../i18n'
+import { ToastContext } from '../../context/Toast'
 
 import { useAllSets } from './useAllSets'
 import { Sets } from './Sets'
 
 const PER_PAGE = 9
 
-export function PageAllSets({ page }) {
+export function CreateNewSetDialog({ onCreateNewSet, onOpenChange, ...props }) {
+  const [name, setName] = useState('')
   const { _e } = useTranslation()
-  const { user } = useUser({ redirectIfUnauthenticated: true })
-  const router = useRouter()
 
+  const updateName = (e) => setName(e.target.value)
+
+  return (
+    <Dialog
+      title={_e('flashcardset.create')}
+      onOpenChange={onOpenChange}
+      {...props}>
+      <form action="" className="p-2">
+        <div className="grid grid-cols-2 gap-2">
+          <Input name="name" value={name} onChange={updateName} />
+          <Button
+            className="border border-sky-300 bg-sky-100 text-sm font-semibold text-sky-700 hover:border-sky-400"
+            onClick={onCreateNewSet(name)}>
+            {_e('common.create')}
+          </Button>
+        </div>
+      </form>
```

Comment:
Should there be some better style? Missing placeholder and label so I don't know what this input wants me to enter.
<img width="531" alt="image" src="https://user-images.githubusercontent.com/172503/165140665-b04307e1-7b35-4a44-9e1c-02758a2fa257.png">